core/src/main/java/overflowdb/NodeRef.java

```diff
@@ -99,9 +99,9 @@ private final synchronized N getSynchronized() throws IOException {
     if (ref != null) {
       return ref;
     } else {
-      final N node = readFromDisk(id);
+      final N node = readFromDisk();
       if (node == null) throw new IllegalStateException("unable to read node from disk; id=" + id);
-      this.node = node;
+      assert(this.node == node);
```

Comment:
Java's `assert`s are not being checked at runtime by default 😢 
I tend to `throw new AssertionError` in this case, but I guess there's other options. 

Also: the error text in this case wouldn't be very helpful.